src/arrayMethodJoin.js

```diff
@@ -5,7 +5,35 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    if (this.length === 1) {
+      return `${this[0]}`;
+    }
+
+    let sepr = separator;
+
+    if (sepr === undefined) {
+      sepr = ',';
+    }
+
+    let result = '';
+
+    if (this.length > 1) {
+      result += typeCheck(this[0]);
+
+      for (let i = 1; i < this.length; i++) {
+        result += `${sepr}${typeCheck(this[i])}`;
+      }
+    }
+
+    function typeCheck(char) {
+      if (char === undefined || char === null) {
+        return '';
+      } else {
+        return `${char}`;
+      }
+    }
+
+    return result;
```

Comment:
let's simplify your solution, you need only 2 checks:
1. if  this[i] is not undefined and isn't null, add it to the result string
2. if it isn't last element, add separator
thats it